src/classes/RD_RecurringDonations_TEST2.cls

```diff
@@ -42,19 +42,25 @@ private with sharing class RD_RecurringDonations_TEST2 {
     private static final Integer NUM_INSTALLMENTS = 12;
 
 
+    /****
+    * @description Creates data required for unit tests
+    */
+    @TestSetup
```

Comment:
Could not reproduce the error in unit tests but cleaned up the test class after deleting the RD batch test executed by a read-only or standard user. Also, could not reproduce the issue on a scratch org. Thus, the current behaviour for legacy and enhanced RD batch jobs can be verified to work as expected and when the PR is merged into the `master`, upgrade the beta package on the dev org where scheduled jobs are run by the "Nonprofit Success Pack" and ensure the RD scheduled job does not generate the `AsyncApexJob` error.

@rponti-sforg , @npsp-reedestockton , @andrewyu-salesforce , @judisohn